core/vm/contracts_lightclient.go

```diff
@@ -135,27 +187,29 @@ func (c *iavlMerkleProofValidate) Run(input []byte) (result []byte, err error) {
 	return result, nil
 }
 
-// tmHeaderValidate implemented as a native contract.
-type tmHeaderValidateNano struct{}
-
-func (c *tmHeaderValidateNano) RequiredGas(input []byte) uint64 {
-	return params.TendermintHeaderValidateGas
-}
-
-func (c *tmHeaderValidateNano) Run(input []byte) (result []byte, err error) {
-	return nil, fmt.Errorf("suspend")
-}
-
-//------------------------------------------------------------------------------------------------------------------------------------------------
-type iavlMerkleProofValidateNano struct{}
-
-func (c *iavlMerkleProofValidateNano) RequiredGas(input []byte) uint64 {
-	return params.IAVLMerkleProofValidateGas
+func forbiddenAbsenceOpVerifier(op merkle.ProofOperator) error {
```

Comment:
This can be done by removing related registers in `multistoreproof.go`

https://github.com/bnb-chain/bsc/blob/1a2da608df5a2e77aedf175225fa3210319f0667/core/vm/lightclient/multistoreproof.go#L131-L138

Reply:
We can not remove this, in case the history block or tx  contains `ProofOpIAVLAbsence` op.